source/uwp/Renderer/lib/TextHelpers.cpp

```diff
@@ -19,21 +19,36 @@ using namespace ABI::Windows::UI::Xaml::Controls;
 using namespace ABI::Windows::UI::Xaml::Media;
 using namespace ABI::Windows::UI::Xaml;
 
-HRESULT SetMaxLines(ITextBlock* textBlock, UINT maxLines)
+HRESULT StyleXamlTextBlockProperties(_In_ ABI::AdaptiveNamespace::IAdaptiveTextBlock* adaptiveTextBlock,
+                                     _In_ ABI::AdaptiveNamespace::IAdaptiveRenderContext* renderContext,
+                                     _In_ ABI::AdaptiveNamespace::IAdaptiveRenderArgs* renderArgs,
+                                     _In_ ITextBlock* xamlTextBlock)
 {
-    ComPtr<ITextBlock> localTextBlock(textBlock);
-    ComPtr<ITextBlock2> xamlTextBlock2;
-    localTextBlock.As(&xamlTextBlock2);
-    RETURN_IF_FAILED(xamlTextBlock2->put_MaxLines(maxLines));
-    return S_OK;
-}
+    boolean wrap;
+    RETURN_IF_FAILED(adaptiveTextBlock->get_Wrap(&wrap));
+    RETURN_IF_FAILED(SetWrapProperties(xamlTextBlock, wrap));
+
+    UINT32 maxLines;
+    RETURN_IF_FAILED(adaptiveTextBlock->get_MaxLines(&maxLines));
+    if (maxLines != MAXUINT32)
+    {
+        ComPtr<ITextBlock> localXamlTextBlock(xamlTextBlock);
+        ComPtr<ITextBlock2> xamlTextBlock2;
+        RETURN_IF_FAILED(localXamlTextBlock.As(&xamlTextBlock2));
+        RETURN_IF_FAILED(xamlTextBlock2->put_MaxLines(maxLines));
+    }
+
+    RETURN_IF_FAILED(SetHorizontalAlignment(adaptiveTextBlock, xamlTextBlock));
+
+    ComPtr<IAdaptiveTextBlock> localAdaptiveTextBlock(adaptiveTextBlock);
+    ComPtr<IAdaptiveTextElement> adaptiveTextElement;
+    RETURN_IF_FAILED(localAdaptiveTextBlock.As(&adaptiveTextElement));
+
+    if (adaptiveTextElement)
```

Comment:
don't COM rules generally state that `S_OK` means out pointer is non-null?

Reply:
Yup, that's a remnant of a previous implementation, removed check.

---
In reply to: [276058184](https://github.com/Microsoft/AdaptiveCards/pull/2679#discussion_r276058184) [](ancestors = 276058184)